packages/remix/src/utils/remix-config.ts

```diff
@@ -1,19 +1,62 @@
-import { joinPathFragments, readProjectConfiguration, Tree } from '@nx/devkit';
+import {
+  joinPathFragments,
+  readProjectConfiguration,
+  Tree,
+  workspaceRoot,
+} from '@nx/devkit';
 import type { AppConfig } from '@remix-run/dev';
+import { createContext, SourceTextModule } from 'vm';
 
 export function getRemixConfigPath(tree: Tree, projectName: string) {
   const project = readProjectConfiguration(tree, projectName);
   if (!project) throw new Error(`Project does not exist: ${projectName}`);
 
-  for (const ext of ['.cjs', '.js']) {
+  for (const ext of ['.mjs', '.cjs', '.js']) {
     const configPath = joinPathFragments(project.root, `remix.config${ext}`);
     if (tree.exists(configPath)) {
       return configPath;
     }
   }
 }
 
+export function getRemixConfigPathFromProjectRoot(
+  tree: Tree,
+  projectRoot: string
+) {
+  let pathToRemixConfig: string;
+  for (const ext of ['.js', '.mjs', '.cjs']) {
+    pathToRemixConfig = joinPathFragments(projectRoot, `remix.config${ext}`);
+    if (tree.exists(pathToRemixConfig)) {
+      return pathToRemixConfig;
+    }
+  }
+
+  throw new Error(
+    `Could not find a Remix Config File. Please ensure a "remix.config.js" file exists at the root of your project.`
+  );
+}
+
+const _remixConfigCache: Record<string, AppConfig> = {};
+
 export async function getRemixConfigValues(tree: Tree, projectName: string) {
-  const remixConfigPath = getRemixConfigPath(tree, projectName);
-  return eval(tree.read(remixConfigPath, 'utf-8')) as AppConfig;
+  const remixConfigPath = joinPathFragments(
+    workspaceRoot,
+    getRemixConfigPath(tree, projectName)
+  );
+  const cacheKey = `${projectName}/${remixConfigPath}`;
+  let appConfig = _remixConfigCache[cacheKey];
+
+  if (!appConfig) {
+    try {
+      const importedConfig = await Function(
+        `return import("${remixConfigPath}")`
```

Comment:
To avoid import cache, you can append a querystring to the end.

e.g. `import("${remixConfigPath}?t=${Date.now()}"`

Otherwise, changes to the file will not reload config in Nx daemon.

We should do this for other plugins as well.

Reply:
Interesting, ok!